### PR TITLE
Tp1 1033 migrate ga to gtm

### PR DESCRIPTION
**Pull request by dlopezvsr**
# Description

This PR updates all code snippets using `ReactGA.event({})` and replace them for `window.dataLayer.push({})` adding also the `event` property to the object, which is listened and handled by GTM to identify the event once is triggered. Also next to each code update, a trigger and tag component were created inside of GTM in order to push the data properly to GA4.

# How to test/review

- Once these changes are deployed into staging we will have to log in to GA and GTM.
- Go to https://analytics.google.com/analytics/web/ > login to dashboard
- Open admin settings and on "Data display" section, > open "Events"
- Search for the new tags added from GTM. 

In GTM:
- Log in to https://tagmanager.google.com
- Once inside the overview section you can press "Preview"
- Add the URL of staging or the one from the review app and connect.
- Test in real-time the interaction.

Link to sample test page: https://foundation-s-tp1-1033-m-xabthl.herokuapp.com/en/
Related PRs/issues:
- https://github.com/MozillaFoundation/foundation.mozilla.org/issues/12599
- https://github.com/MozillaFoundation/foundation.mozilla.org/issues/12696

# Checklist

**Tests**
- [ ] Is the code I'm adding covered by tests?

**Changes in Models:**
- [ ] Did I update or add new fake data?
- [ ] Did I squash my migration?
- [ ] Are my changes [backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?

**Documentation:**
- [ ] Is my code documented?
- [ ] Did I update the READMEs or wagtail documentation?

**Merge Method**
**💡❗Remember to use squash merge when merging non-feature branches into `main`**

┆Issue is synchronized with this [Jira Story](https://mozilla-hub.atlassian.net/browse/TP1-1082)

### CODE REVIEW

File: network-api/networkapi/templates/pages/base.html [hunk 1]
```diff
@@ -39,9 +39,9 @@
 
         {% block ga_identifier %}
             <meta name="google-site-verification" content="D7k-r3fHm-XfJ9E7T1uZ5aqHJG2mx-0uUZFeBUDN2lY">
-            <meta name="ga-identifier" content="UA-87658599-6">
         <!-- The current GTM ID replaced an unkown GTM container with no clear access -->
-            <meta name="gtm-identifier" content="GTM-PQQ8H6PM">
+            <<meta name="gtm-identifier" content="GTM-PQQ8H6PM">
```
[the remark this answers is not in the document]
Comment (dlopezvsr) [reply network-api/networkapi/templates/pages/base.html]: Great catch @ramram-mf, I just corrected on this last commit. Thank you! 😄

[the remark this answers is not in the document]
Comment (ramram-mf) [reply network-api/networkapi/templates/pages/base.html]: Thanks. Let's QA this!

**Review by ramram-mf — APPROVED** [review]
LGTM. Thanks Diego

**Review by danielfmiranda — CHANGES_REQUESTED** [review]
Hi @dlopezvsr !

Great work on this PR, the code changes are looking good and I appreciate the effort you've put into this!

There's just one small adjustment needed regarding the inline scripts.

We use a nonce in our CSP settings to ensure that only scripts we explicitly allow can run on our pages. 

I noticed that the scripts added don't yet include a nonce. To maintain our security standards, could you please update these to include it? This would look something like:

```Javascript
<script nonce="{{ request.csp_nonce }}">window.dataLayer = window.dataLayer || [];</script>
```



Let me know if you have any questions or need further clarification. I'm happy to help!

**Comment by dlopezvsr** [comment]
@danielfmiranda thank you so much for the very important observation, I just pushed a new commit with the `request.csp_nonce` template inside of the inline script. 

Let me know in case you have other comments 🙌 😄

**Review by danielfmiranda — APPROVED** [review]
LGTM!

**Pull request merged**